src/backend/aspen/app/views/sample.py

```diff
@@ -26,12 +26,15 @@ def _format_created_date(sample: Sample) -> str:
 
 def _format_gisaid_accession(
     sample: Sample, entity_id_to_accession_map: Mapping[int, Accession]
-) -> str:
+) -> Mapping[str, Union[str, None]]:
     uploaded_entity = sample.get_uploaded_entity()
     accession = entity_id_to_accession_map.get(uploaded_entity.entity_id, None)
+    if sample.czb_failed_genome_recovery:
+        # todo need to add the private option here for v3 a user uploads and flags a private sample
+        return {"status": "not_eligible", "gisaid_id": None}
     if accession is not None:
-        return accession.public_identifier
-    return "NOT SUBMITTED"
+        return {"status": "accepted", "gisaid_id": accession.public_identifier}
+    return {"status": "rejected", "gisaid_id": None}
```

Comment:
@ttung should a sample with an uploaded_entity (for GISAID) but no accession be considered status PENDING?

Reply:
Or should we change this flag to be "NO_INFO"

Reply:
As per our discussions:

1. A GISAID submission workflow that started less than X time ago but hasn't shown up in the GISAID dump is considered pending.
2. A GISAID submission workflow that started more than X time ago but hasn't shown up in the GISAID dump is considered failled.

Reply:
Ah ok so would the logic be:
- query all workflows for the given `uploaded_entity`
- check if any are PUBLIC_REPOSITORY_SUBMISSION and have the WorkflowStatusType STARTED
- do a date check....what should the date check be? A week?

Reply:
I think the long-term logic is:

1. Retrieve all the workflows coming off of this entity that are of type `AccessionWorkflow`, as well as the entities that are the outputs from those workflows.
2. If there are no workflows, then it was never submitted.
3. If there are failed workflows, then the submission has failed.  There should be a separate mechanism to mark workflows failed if they are too old.

Reply:
Shorter term, we may need to think things through...